beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestHardenCatalogPlugin.java

```diff
@@ -265,25 +259,19 @@ public void reset() {
         }
 
         private Iterable<StandalonePluginCatalog> toStandalonePluginCatalogs(final List<StaticCatalog> input) {
-            return Iterables.transform(input, new Function<StaticCatalog, StandalonePluginCatalog>() {
-                @Override
-                public StandalonePluginCatalog apply(final StaticCatalog input) {
-
-                    final StandaloneCatalog standaloneCatalog = (StandaloneCatalog) input;
-                    return new TestModelStandalonePluginCatalog(new DateTime(input.getEffectiveDate()),
-                                                                ImmutableList.copyOf(standaloneCatalog.getSupportedCurrencies()),
-                                                                ImmutableList.<Product>copyOf(standaloneCatalog.getProducts()),
-                                                                ImmutableList.<Plan>copyOf(standaloneCatalog.getPlans()),
-                                                                standaloneCatalog.getPriceLists().getDefaultPricelist(),
-                                                                ImmutableList.<PriceList>copyOf(standaloneCatalog.getPriceLists().getChildPriceLists()),
-                                                                standaloneCatalog.getPlanRules(),
-                                                                null /* ImmutableList.<Unit>copyOf(input.getCurrentUnits()) */);
-                }
-
-                private <I, C extends I> List<I> listOf(@Nullable C[] input) {
-                    return (input != null) ? ImmutableList.<I>copyOf(input) : ImmutableList.<I>of();
-                }
-            });
+            return input.stream()
+                    .map(staticCatalog -> {
+                        final StandaloneCatalog standaloneCatalog = (StandaloneCatalog) staticCatalog;
+                        return new TestModelStandalonePluginCatalog(new DateTime(staticCatalog.getEffectiveDate()),
+                                                                    List.of(standaloneCatalog.getSupportedCurrencies()),
+                                                                    List.copyOf(standaloneCatalog.getProducts()),
```

Comment:
`of` vs `copyOf` on purpose?

Also, in the past, you've used `Iterables.toUnmodifiableList` (https://github.com/killbill/killbill/pull/1697#discussion_r876986821). Just wondering if we are consistently using the same patterns?

Reply:
@pierre on that lines:
- standaloneCatalog.getSupportedCurrencies() return an array
- standaloneCatalog.getProducts() return a `Collection`

In `of` vs `copyOf` : because not like Guava's `ImmutableList.copyOf` that have wide range of target parameter (array, `Collection`, `Iterator`, etc), `List.copyOf` only applied to `Collection` (not even `Iterable`). If we want to apply to array, we should use `List.of`. None of List methods accept `Iterables`, so `Iterables.toUnmodifiableList()` (or `toList()) comes as helper.

That's why we have different `List` methods applied to parameter values.

Reply:
Got it, thanks!